torchci/docs/architecture.md

```diff
@@ -0,0 +1,82 @@
+# `torchci` architecture notes
+## Primary write path
```

Comment:
A quick question: is this correct that all the webhook events we support are in [webhookToDynamo.ts](https://github.com/pytorch/test-infra/blob/main/torchci/lib/bot/webhookToDynamo.ts#L139-L145)?

Reply:
Yes. You can add more but those are the ones we currently use in the HUD.

We also save the all raw webhook payloads to an S3 bucket
(ossci-raw-job-status), so if we need to backfill a new webhook event we
can do that as well.

Reply:
I see. Your note about `ossci-raw-job-status` is actually quite important to know

Reply:
let me add it as a note in the doc